### ISSUE TITLE
Quantile regression

### ISSUE THREAD

**Issue by StatPal** [mlverse/torch#845]
Is there any immediate way to incorporate Quantile Loss (and consequently Quantile regression) in torch with R? 
I have found something in PyTorch here: https://github.com/pytorch/pytorch/issues/38035. 

This issue is opened as a feature request, but I can try to contribute if it is not already done.
Thanks.

**Comment by dfalbel** [comment]
HI @StatPal !

There's no built-in quantile loss function in torch but you can implement one with something like:

```
quantile_loss <- torch::nn_module(
  initialize = function(quantiles) {
    self$quantiles <- torch::torch_tensor(sort(quantiles))$unsqueeze(1)$unsqueeze(1)
  },
  forward = function(y_pred, y_true) {
    other <- torch::torch_zeros_like(y_pred)
    error <- y_true - y_pred

    low_res <- torch::torch_max(error, other = other)
    up_res <- torch::torch_max(-error, other = other)

    quantiles <- self$quantiles$to(device = y_true$device)
    torch::torch_mean(quantiles * low_res + (1 - quantiles) * up_res)
  }
)
```

For instance this is what's used [here](https://github.com/mlverse/tft/blob/main/R/tft-layers.R#L107-L121)

**Comment by StatPal** [comment]
Thank you very much for your suggestion.
